internal/codespaces/connection/tunnels_api_server_mock.go

```diff
@@ -0,0 +1,444 @@
+			// Use regex to check if the path ends with a number
+			match, err := regexp.MatchString(`\/\d+$`, r.URL.Path)
+			if err != nil {
+				log.Fatalf("regexp.MatchString returned an error: %v", err)
+			}
+
+			// If the path ends with a number, it's a request for a specific port
+			if match || r.Method == http.MethodPost {
+				if r.Method == http.MethodDelete {
+					w.WriteHeader(http.StatusOK)
+					return
+				}
+
+				tunnelPort := &tunnels.TunnelPort{
+					AccessControl: &tunnels.TunnelAccessControl{
+						Entries: []tunnels.TunnelAccessControlEntry{},
+					},
+				}
+
+				// Convert the tunnel to JSON and write it to the response
+				response, err = json.Marshal(*tunnelPort)
+				if err != nil {
+					log.Fatalf("json.Marshal returned an error: %v", err)
+				}
+			} else {
+				// If the path doesn't end with a number and we aren't making a POST request, return an array of ports
+				tunnelPorts := []tunnels.TunnelPort{
+					{
+						AccessControl: &tunnels.TunnelAccessControl{
+							Entries: []tunnels.TunnelAccessControlEntry{},
+						},
+					},
+				}
+
+				response, err = json.Marshal(tunnelPorts)
+				if err != nil {
+					log.Fatalf("json.Marshal returned an error: %v", err)
+				}
+			}
+
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		// Write the response
+		_, _ = w.Write(response)
+	}))
+
+	url, err := url.Parse(mockServer.URL)
+	if err != nil {
+		return nil, fmt.Errorf("url.Parse returned an error: %w", err)
+	}
+	return &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyURL(url),
+		},
+	}, nil
+}
+
+const sshPrivateKey = `-----BEGIN RSA PRIVATE KEY-----
+MIICXgIBAAKBgQC6VU6XsMaTot9ogsGcJ+juvJOmDvvCZmgJRTRwKkW0u2BLz4yV
+rCzQcxaY4kaIuR80Y+1f0BLnZgh4pTREDR0T+p8hUsDSHim1ttKI8rK0hRtJ2qhY
+lR4qt7P51rPA4KFA9z9gDjTwQLbDq21QMC4+n4d8CL3xRVGtlUAMM3Kl3wIDAQAB
+AoGBAI8UemkYoSM06gBCh5D1RHQt8eKNltzL7g9QSNfoXeZOC7+q+/TiZPcbqLp0
+5lyOalu8b8Ym7J0rSE377Ypj13LyHMXS63e4wMiXv3qOl3GDhMLpypnJ8PwqR2b8
+IijL2jrpQfLu6IYqlteA+7e9aEexJa1RRwxYIyq6pG1IYpbhAkEA9nKgtj3Z6ZDC
+46IdqYzuUM9ZQdcw4AFr407+lub7tbWe5pYmaq3cT725IwLw081OAmnWJYFDMa/n
+IPl9YcZSPQJBAMGOMbPs/YPkQAsgNdIUlFtK3o41OrrwJuTRTvv0DsbqDV0LKOiC
+t8oAQQvjisH6Ew5OOhFyIFXtvZfzQMJppksCQQDWFd+cUICTUEise/Duj9maY3Uz
+J99ySGnTbZTlu8PfJuXhg3/d3ihrMPG6A1z3cPqaSBxaOj8H07mhQHn1zNU1AkEA
+hkl+SGPrO793g4CUdq2ahIA8SpO5rIsDoQtq7jlUq0MlhGFCv5Y5pydn+bSjx5MV
+933kocf5kUSBntPBIWElYwJAZTm5ghu0JtSE6t3km0iuj7NGAQSdb6mD8+O7C3CP
+FU3vi+4HlBysaT6IZ/HG+/dBsr4gYp4LGuS7DbaLuYw/uw==
+-----END RSA PRIVATE KEY-----`
+
+type RelayServer struct {
+	httpServer  *httptest.Server
+	errc        chan error
+	sshConfig   *ssh.ServerConfig
+	channels    map[string]channelHandler
+	accessToken string
+
+	serverConn *ssh.ServerConn
+}
+
+type RelayServerOption func(*RelayServer)
+type channelHandler func(context.Context, ssh.NewChannel) error
+
+func NewRelayServer(opts ...RelayServerOption) (*RelayServer, error) {
```

Comment:
Similar suggestion to above.

```suggestion
func NewMockRelayServer(opts ...RelayServerOption) (*RelayServer, error) {
```